### ISSUE TITLE
Typechecker improvements

### ISSUE THREAD

**Issue by iccir** [musictheory/NilScript#173]
A lot of this work is already done, but I wanted to make an issue to document it.

- `check-types` should work with multiple `Compiler` objects when the `Compiler#uses` API is used. Historically, for performance reasons, we have recommended a single aggregate `Compiler` with all source files. This fails to catch certain types of dependency graph issues.
- We should use `worker_threads` to execute multiple copies of the TypeScript compiler in parallel.
- We should use the Language Services API and let TypeScript handle incremental compile logic.
- There should be an option to prevent `Compiler#compile` from waiting on type checker results. This would deliver the performance wins of the aggregate type checker while also catching the above-mentioned dependency graph issues.
